### PR TITLE
Add Dialect API Post call for integration

### PR DESCRIPTION
**Pull request by echohtp**

### CODE REVIEW

File: crates/indexer/src/geyser/client.rs [hunk 1]
```diff
@@ -92,4 +95,34 @@ impl Client {
...
+            .new()
+            .post("")
+            .body(Message {
+                msgType: MessageType::NftOffer,
+                data: MessageData {
+                    address: "1111111111111111111111".to_string(),
+                },
+            })
+            .send()
+            .await?;
```
Comment (kespinola) [inline crates/indexer/src/geyser/client.rs:126]: http is misleading for this client. its actually a rabbitmq client.

you aren't looking to add another producer. its something more like the db but preconfigured http_sender.

Comment (echohtp) [reply crates/indexer/src/geyser/client.rs:126]: hm, okay :+1:

File: crates/indexer/src/geyser/accounts/receipt.rs [hunk 2]
```diff
@@ -120,15 +121,23 @@ pub(crate) async fn process_bid_receipt(
...
+            ))
+            .get_result::<bool>(db);
             insert_into(bid_receipts::table)
                 .values(&row)
                 .on_conflict(bid_receipts::address)
                 .do_update()
                 .set(&row)
-                .execute(db)
+                .execute(db)?;
+            if Ok(true) == bid_receipt_exists || row.purchase_receipt.is_some() {
```
Comment (ray-kast) [inline crates/indexer/src/geyser/accounts/receipt.rs:134]: This if statement does nothing, `return` in Rust closures is more like `next` in Ruby closures.

**Review by ray-kast — CHANGES_REQUESTED** [review]
What are the timings on the Dialect POST requests?  If it's too slow we may need a separate HTTP indexer for that.

File: crates/indexer/src/geyser/accounts/receipt.rs (hunk 2 above)
Comment (echohtp) [reply crates/indexer/src/geyser/accounts/receipt.rs:134]: whats the suggestion here @ray-kast ?

**Review by ray-kast — CHANGES_REQUESTED** [review]
Something has gone horribly wrong here.

**Comment by ray-kast** [comment]
Gonna reopen this with a less broken Git history.

**Pull request closed without merging**